test/suites/secondary_url.bash

```diff
@@ -0,0 +1,41 @@
+SUITE_secondary_url_SETUP() {
+    generate_code 1 test.c
+}
+
+SUITE_secondary_url() {
+    # -------------------------------------------------------------------------
+    TEST "Reject empty url (without config attributes)"
+
+    export CCACHE_SECONDARY_STORAGE="|"
+    $CCACHE_COMPILE -c test.c 2> "${ABS_TESTDIR}/stderr.log"
+    expect_contains "${ABS_TESTDIR}/stderr.log" "must provide an URL"
+
+    # -------------------------------------------------------------------------
+#    TEST "Reject empty url (but with config attributes)"
+#
+#    export CCACHE_SECONDARY_STORAGE="|key=value"
+#    $CCACHE_COMPILE -c test.c 2> "${ABS_TESTDIR}/stderr.log"
+#    expect_contains "${ABS_TESTDIR}/stderr.log" "must provide an URL"
+
+    # -------------------------------------------------------------------------
+    TEST "Reject invalid url"
+
+    export CCACHE_SECONDARY_STORAGE="://qwerty"
+    $CCACHE_COMPILE -c test.c 2> "${ABS_TESTDIR}/stderr.log"
+    expect_contains "${ABS_TESTDIR}/stderr.log" "Cannot parse URL"
+
+    # -------------------------------------------------------------------------
+    TEST "Reject missing scheme"
+
+    export CCACHE_SECONDARY_STORAGE="/qwerty"
+    $CCACHE_COMPILE -c test.c 2> "${ABS_TESTDIR}/stderr.log"
+    expect_contains "${ABS_TESTDIR}/stderr.log" "URL scheme must not be empty"
+
+    # -------------------------------------------------------------------------
+    TEST "Redact user info"
+
+    export CCACHE_SECONDARY_STORAGE="file://username:secret@host$PWD/secondary"
+    $CCACHE_COMPILE -c test.c
+    expect_contains "${CCACHE_LOGFILE}" "username:REDACTED"
+    #expect_not_contains "${CCACHE_LOGFILE}" "secret" # --show-config leaks the secret
```

Comment:
Fixing this requires moving the parsing code into the configuration class. Should we do this now or could we postpone it?

Reply:
Sounds reasonable. I was inspired by [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1) which recommends:
> Applications should not render as clear text any data after the first colon (":") character found within a userinfo

Reply:
Redacting secrets in logs has been addressed in 873384667a5e8ba0e688303e18f34196ab3b57c6.